### PR TITLE
navigate to profile from public chat photo

### LINKED ISSUE #5184: Add possibility to navigate to user profile from public chat and go back
### User Story

As a user, I want to initiate chat  with public chat participants

### Description

*Type*: Feature
*Summary*: user should be able to open profile public chat member by clicking on his profile picture and go back.

#### How it is implemented in mobile

[comment]: # (Describe what you expected to happen.)
When user clicks on user pic of any public chat participant, he is redirectied to:
lit
The arrow in left top corner is redirecting user back to public chat.

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

- [ ] add redirect to user profile from public chat
- [ ] add way for user to get back to public chat

### PR DESCRIPTION
**Pull request by rcullito**
fixes #5184

### CODE REVIEW

**Comment by rcullito** [comment]
TODO
1. move event to dedicated desktop namespace and make sure this is required correctly
2. current gyfcat implementation for subscription get-current-chat-name assumes a contact if 1-1. work this out for when we're still dealing with public chat person.

**Review by vkjr — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/chat/views.cljs [hunk 1]
```diff
@@ -189,30 +194,41 @@
...
+                 (let [{:keys [pending? whisper-identity] :as contact} (or current-chat-contact
+                                                                           already-stored-contact
+                                                                           (utils.contacts/whisper-id->new-contact contact-identity))]
+                   [react/view
+                    [react/view {:style styles/desktop-profile-back-button}
+                     toolbar.view/default-nav-back]
+                    [profile.views/profile-badge contact]
+                    [react/view {:style styles/chat-profile-body}
+                     ;; pending? will come back as nil for public chat so explicitly check for false here
+                     (if (false? pending?)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/desktop/main/chat/views.cljs]: `if-not` ?

File: src/status_im/ui/screens/desktop/main/chat/views.cljs [hunk 2]
```diff
@@ -189,30 +194,41 @@
...
-                   [react/text {:style styles/chat-profile-contact-code} (i18n/label :t/contact-code)]
-                   [react/text {:style      {:font-size 14}
-                                :selectable true} public-key]]]))
+                                    :accessibility-label :send-message-link}
+                        [vector-icons/icon :icons/chats {:style (styles/chat-profile-icon colors/blue)}]]
+                       [react/text {:style (styles/contact-card-text colors/blue)}
+                        (i18n/label :t/send-message)]]]
+                     [react/text {:style styles/chat-profile-contact-code} (i18n/label :t/contact-code)]
+                     [react/text {:style      {:font-size 14}
+                                  :selectable true} whisper-identity]]])))
```
Comment (flexsurfer) [inline src/status_im/ui/screens/desktop/main/chat/views.cljs]: better to move `whisper-identity` to the next line, more readable

File: src/status_im/ui/screens/desktop/main/chat/views.cljs [hunk 3]
```diff
@@ -56,9 +58,12 @@
        [react/text {:style styles/author} name]])))
 
 (views/defview member-photo [from]
-  [react/view
-   [react/image {:source {:uri (identicon/identicon from)}
-                 :style  styles/photo-style}]])
+  [react/touchable-highlight {:on-press #(do
+                                           (re-frame/dispatch [:set-public-chat-whisper-identity from])
```
Comment (flexsurfer) [inline src/status_im/ui/screens/desktop/main/chat/views.cljs]: we're trying to do not dispatch more than one event, so better to create a new event and change app-db directly

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/chat/views.cljs (hunk 3 above)
Comment (rcullito) [reply src/status_im/ui/screens/desktop/main/chat/views.cljs]: Awesome, thanks for this feedback @flexsurfer. I've updated to act on `app-db` directly.

File: src/status_im/ui/screens/desktop/main/chat/views.cljs (hunk 1 above)
Comment (rcullito) [reply src/status_im/ui/screens/desktop/main/chat/views.cljs]: Thanks for bringing this into the light. I *think* a check for an explicit false is appropriate here as the desired behavior is to have `nil` handled the same as `true`. (i.e. both indicative of pending? not having been manually updated to false.) 

I've added more comments inline to address the intention here. Let me know if something still seems off.

**Comment by churik** [comment]
@rcullito 
1. User is navigating to profile from public chat only after double-clicking on user picture.
Expected that he is navigating right after click on user picture.
Video is [here](http://take.ms/m0K2N)
2. I noticed that %CPU is increased when you are switching to profile and back.
Video is [here](http://take.ms/v6Fsl).
Let me know if it is related to https://github.com/status-im/status-react/issues/5271

**Comment by rcullito** [comment]
@churik 

regarding double click issue I've spoken with @MaxRis and believe this should be addressed by his current PR: https://github.com/status-im/react-native-desktop/pull/301 in react-native-desktop. 

CPU usage seemed a continuation of general CPU spiking when loading chat messages. I found that the spike in this PR correlated more with loading the messages than it did loading of the profile, so I think we can say it is bundled in with #5271 as you thought. One hypothesis is that this might be a result of how Whisper works beneath the surface, requiring proof of work? Do you think this is possible @yenda? Or am I totally taking a comment from our chat today out of context? :smile_cat:

**Comment by churik** [comment]
@rcullito 
in this case please rebase it on `develop` branch and nmove `TO TEST` column once https://github.com/status-im/react-native-desktop/pull/301 will be tested and merged to `react-native` repo

**Comment by churik** [comment]
@rcullito any progress on this?
Just FYI: to rebase this, you need to create new PR based on `develop` branch and move it to `TO TEST` column.

**Comment by vitvly** [comment]
@Churik @vkjr Can we close this as it is superseded by #5546?

**Comment by vkjr** [comment]
@siphiuel, I think yes.

**Comment by churik** [comment]
@siphiuel I believe yes

**Pull request closed without merging**